.github/workflows/common-api-ci.yml

```diff
@@ -0,0 +1,201 @@
+          - 9200/tcp
+        options: --health-cmd="curl http://localhost:9200/_cluster/health" --health-interval=10s --health-timeout=5s --health-retries=10
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+        # If a particular API project needs to do some special docker setup
+        # (e.g. Sitewide Search sets up synonyms for Elasticsearch), it should
+        # be encapsulated in a docker-setup.sh script.
+      - name: project-specific Docker setup.
+        run: |
+            if [[ -e 'integration-tests/bin/docker-setup.sh' ]]; then
+              CONTAINER_ID=$(docker ps -q --filter "ancestor=elasticsearch:7.17.5")
+              integration-tests/bin/docker-setup.sh $CONTAINER_ID
+            fi
+
+      - name: Download Published Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.artifact-name }}-api-ubuntu-latest
+          path: built-api
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          global-json-file: global.json
+          source-url: "https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json"
+
+      - name: Load Data into elasticsearch & prepare for tests
+        env:
+          ELASTIC_SEARCH_HOST: http://localhost:${{ job.services.elasticsearch.ports[9200] }}
+        run: |
+              ## Create test output and API logging location
+              mkdir -p integration-tests/target
+
+              ## Load the elasticsearch data
+              ./integration-tests/bin/load-integration-data.sh
+
+      - name: Start API
+        env:
+          Elasticsearch__Servers: http://localhost:${{ job.services.elasticsearch.ports[9200] }}
+          ASPNETCORE_LOGGING__CONSOLE__DISABLECOLORS: true
+          API_URL: http://localhost:5000
+          SLEEP_TIMEOUT: 5
+          WAIT_DURATION: 120
+          APP_PATH: ${{ github.workspace }}/built-api/${{ inputs.artifact-name }}-api
+        run: |
+              ## Start the app and log output
+              ## NOTE: We must change directory because in order for dotnet to be able to locate the appsettings.json file.
+              cd $APP_PATH && ${{ github.workspace }}/integration-tests/bin/start-api.sh > ${{ github.workspace }}/integration-tests/target/api_log.txt 2>&1 &
+
+              time_waited=1
+              echo "Checking status of ${API_URL}."
+              until $(curl --output /dev/null --silent --fail ${API_URL}); do
+                echo $?
+
+                if [ $time_waited -ge $WAIT_DURATION ]; then
+                    echo "Waited past duration. Exiting"
+                    cat integration-tests/target/api_log.txt
+                    exit 1
+                fi
+
+                sleep $SLEEP_TIMEOUT
+                time_waited=$((time_waited + SLEEP_TIMEOUT))
+              done
+
+              echo "API is up"
+
+      - name: Run Integration Test
+        id: integration_tests
+        continue-on-error: true
```

Comment:
I think that `continue-on-error` [can mess with whether your job](https://stackoverflow.com/questions/62045967/is-there-a-way-to-continue-on-error-while-still-getting-correct-feedback#comment137995684_73357322) is marked as `failed` or not. It might also make the step outcome always be `success`? 

I'd suggest removing it, marking the upload step below as `if: ${{ always() }}` and removing the last step.

Reply:
You don't want the `continue-on-error` anymore now, because `always()` will fire anyway.